tests/ingest_runs/test_genomic_data_loader.py

```diff
@@ -0,0 +1,299 @@
+    """
+    loader = GenomicDataLoader(FAKE_STUDY)
+    manifest_df = study_generator.dataframes['gwo_manifest.tsv']
+    gfs = loader.load_harmonized_genomic_files(manifest_df)
+    loader.load_specimen_harmonized_gf_links(gfs)
+    mock_s3_scrape.assert_called_once()
+    assert mock_load_entities.call_count == 2
+    # Now check the biospecimen gf links loading
+    entity_type2, df2 = mock_load_entities.call_args_list[1][0]
+    assert entity_type2 == BIO_GEN_FILE
+    assert isinstance(df2, pd.DataFrame)
+    assert not df2.empty
+    manifest_df = manifest_df.drop_duplicates(["KF Participant ID"])
+    assert manifest_df.shape[0] == df2.shape[0]
+    expected_columns = {
+        CONCEPT.GENOMIC_FILE.ID,
+        CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID,
+        CONCEPT.GENOMIC_FILE.TARGET_SERVICE_ID,
+        CONCEPT.BIOSPECIMEN.ID,
+        CONCEPT.BIOSPECIMEN_GENOMIC_FILE.VISIBLE,
+    }
+    for col in expected_columns:
+        assert col in df2
+    assert df2.shape[1] == len(expected_columns)
+
+
+def test_get_seq_experiment_genomic_files(
+    study_generator, mock_get_entities
+):
+    """
+    Test the _get_seq_experiment_genomic_files_ function. This can be done in
+    isolation.
+    """
+    loader = GenomicDataLoader(FAKE_STUDY)
+    output = loader._get_seq_experiment_genomic_files()
+    # Check _get_entities_ was called the right number of times and the output
+    # is as we expect
+    assert mock_get_entities.call_count == 3
+    expected_columns = {
+        CONCEPT.SEQUENCING.TARGET_SERVICE_ID,
+        CONCEPT.SEQUENCING.CENTER.TARGET_SERVICE_ID,
+        CONCEPT.SEQUENCING.ID,
+        CONCEPT.GENOMIC_FILE.SOURCE_FILE,
+    }
+    for col in expected_columns:
+        assert col in output
+    assert output.shape[1] == len(expected_columns)
+    gfs = study_generator.dataservice_payloads[GEN_FILE]
+    harm_gf = [
+        key for key, value in gfs.items() if value["is_harmonized"] == "False"
+    ]
+    assert output.shape[0] == len(harm_gf)
+
+
+def test_load_seq_exp_harmonized_genomic_files(
+    study_generator, mock_s3_scrape, mock_load_entities, mock_get_entities,
+    mock_validate
+):
+    """
+    Test the _load_seq_exp_harmonized_genomic_files_ function. Here no steps of
+    the ingest process are mocked, so this is really testing the entire
+    process.
+    """
+    loader = GenomicDataLoader(FAKE_STUDY)
+    manifest_df = study_generator.dataframes['gwo_manifest.tsv']
+    output_df = loader.ingest_gwo(manifest_df)
+    mock_s3_scrape.assert_called_once()
+
+    # Check that _load_entities_ was called the right number of times
+    assert mock_load_entities.call_count == 3
+    # Check sequencing-experiment-genomic-files for the harmonized
+    # files are loaded properly
+    entity_type, df = mock_load_entities.call_args_list[2][0]
+    assert entity_type == SEQ_EXP_GEN_FILE
+    assert isinstance(df, pd.DataFrame)
+    manifest_df = manifest_df.drop_duplicates(["KF Participant ID"])
```

Comment:
Same comment as above. We want to check that every harmonized gf got linked to one unique sequencing experiment.